src/main/java/com/premiumminds/persistence/MultiplePersistenceTransaction.java

```diff
@@ -79,14 +92,22 @@ public void end() {
 	public void setRollbackOnly() {
 		log.debug("Marking application transaction for rollback");
 		for(PersistenceTransaction pt : persistenceTransactions){
-			pt.setRollbackOnly();
+			try {
+				pt.setRollbackOnly();
+			} catch (Exception e) {
+				log.warn("Failed to mark transaction "+pt.getClass().getSimpleName()+" for rollback: "+e.getMessage());
+			}
 		}
 	}
 
 	@Override
 	public boolean isRollbackOnly() {
 		for(PersistenceTransaction pt : persistenceTransactions){
-			if(pt.isRollbackOnly()) return true;
+			try {
+				if(pt.isRollbackOnly()) return true;
+			} catch (Exception e) {
+				log.warn("Failed to determine rollback status for transaction "+pt.getClass().getSimpleName()+" for rollback: "+e.getMessage());
```

Comment:
I think that if this happens, we should always say this transaction needs to be rollbacked